src/Bicep.Core/Analyzers/Linter/LinterRule.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Configuration;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    public abstract class LinterRule : IBicepAnalyzerRule
+    {
+        public LinterRule(string code, string ruleName, string description, string docUri,
```

Comment:
>code [](http://example.com/codeflow?start=33&length=4)

Do we want to tie a linter rule to a single violation? If a rule needs to do expensive processing on the semantic model, the authors will need to figure out a way to share state across rule instances.

Something like the unused {variable|parameter} rule could be combined into one as well since it's so similar.

Reply:
There is an upside of the current approach though. Reading configuration and many other things become much simpler.

Reply:
Look at optimization of multiple rules that use find references. Perhaps a better method for doing bulk reference lookup.

Reply:
Tracked in https://github.com/Azure/bicep/issues/2771